widgets/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/validators/RunOnceEditorValidator.java

```diff
@@ -35,17 +36,41 @@ public boolean isValid() {
     if ( null == editor.getStartDate() ) {
       isValid = false;
     } else {
+
       final DateTimeFormat format = DateTimeFormat.getFormat( "MM-dd-yyyy" ); //$NON-NLS-1$
       final String date = format.format( editor.getStartDate() );
-      final String dateTime = date + " " + editor.getStartTime(); //$NON-NLS-1$
+      String time = editor.getStartTime();  //format of time is "hh:mm:ss a"
+
+      if (editor.getStartDate().before( new Date() )){
+       isValid=false;
+      } else {
+
+        if(time.endsWith( TimeUtil.TimeOfDay.AM.toString() ) || time.endsWith( (TimeUtil.TimeOfDay.PM.toString() ) )){
+          time = normalizeTime( time );
+        }
```

Comment:
Test for presence of AM/PM is already done inside `normalizeTime`. There's no need to repeat, afaict.